src/Transaction/ITransaction.php

```diff
@@ -7,34 +7,61 @@
 interface ITransaction
 {
 	/**
+	 * Adds new callback into transaction.
+	 * First parameter in EntityManagerInterface.
+	 * Second is value returned from first (initial) callback.
+	 * Another parameters are passed by calling ITransaction::run(...args).
+	 *
 	 * @param callable $callback
 	 *
 	 * @return \SixtyEightPublishers\DoctrinePersistence\Transaction\ITransaction
 	 */
 	public function then(callable $callback): ITransaction;
 
 	/**
+	 * This callback is called when current transaction successfully ends, but everything may not be committed if current transaction is wrapped into another transaction.
+	 * The only parameter is value returned from first (initial) callback.
+	 *
 	 * @param callable $callback
 	 *
 	 * @return \SixtyEightPublishers\DoctrinePersistence\Transaction\ITransaction
 	 */
 	public function done(callable $callback): ITransaction;
 
 	/**
+	 * This callback is invoked when specified Exception is thrown.
+	 * The only parameter is caught exception.
```

Comment:
Maybe _*caught en exception_?

Reply:
eh .. or *"an caught exception"* maybe?

dunno, I should improve my English :smile:

Reply:
You're right, but _The only parameter is a caught exception._ 😄

Reply:
Thank you for improving my English 🙂